libs/wingc/src/jsify.rs

```diff
@@ -1602,34 +1602,57 @@ impl<'a> FieldReferenceVisitor<'a> {
 
 			ExprKind::Reference(Reference::InstanceMember { object, property }) => {
 				let obj_type = object.evaluated_type.borrow().unwrap();
-				let var = if let Some(cls) = obj_type.as_class_or_resource() {
-					Some(
+				let var = match &*obj_type {
+					Type::Void => unreachable!("cannot reference a member of void"),
+					Type::Function(_) => unreachable!("cannot reference a member of a function"),
+					Type::Optional(_) => unreachable!("cannot reference a member of an optional"),
```

Comment:
Could we avoid these panics? Maybe the void one doesn't matter, but the other two seem like things that a user could type (lmk if I'm wrong) and we probably shouldn't crash the compiler for that.

Reply:
I don't think these are lines are reachable, at least from my time playing around on the branch a bit. Unless we can produce a case where these paths are possible, I don't think we should add a diagnostic since we wouldn't be able to test it

Reply:
Yeah I was thinking about the case of an LSP where incomplete input could possibly lead here, but even if that were true we don't jsify if there's an error in the previous phase (or at all, in the LSP).